packages/core/test/RoutingNode.test.ts

```diff
@@ -40,9 +44,53 @@ const preSendFunction1 = async function (
 	return requestOptions;
 };
 
+const getExecuteSingleFunctions = (
+	workflow: Workflow,
+	runExecutionData: IRunExecutionData,
+	runIndex: number,
+	node: INode,
+	itemIndex: number,
+) =>
+	mock<executionContexts.ExecuteSingleContext>({
+		getItemIndex: () => itemIndex,
+		getNodeParameter: (parameterName: string) =>
+			workflow.expression.getParameterValue(
+				get(node.parameters, parameterName),
+				runExecutionData,
+				runIndex,
+				itemIndex,
+				node.name,
+				[],
+				'internal',
+				{},
+			),
+		getWorkflow: () => ({
+			id: workflow.id,
+			name: workflow.name,
+			active: workflow.active,
+		}),
+		helpers: mock<IExecuteSingleFunctions['helpers']>({
+			async httpRequest(
+				requestOptions: IHttpRequestOptions,
+			): Promise<IN8nHttpFullResponse | IN8nHttpResponse> {
+				return {
+					body: {
+						headers: {},
+						statusCode: 200,
+						requestOptions,
+					},
+				};
+			},
+		}),
+	});
+
 describe('RoutingNode', () => {
 	const additionalData = mock<IWorkflowExecuteAdditionalData>();
 
+	beforeAll(() => {
+		nock.disableNetConnect();
+	});
```

Comment:
Is this necessary? Without it the tests still pass and I don't see how `getExecuteSingleFunctions` uses the network.

Reply:
it's not required. but since `RoutingNode` is primarily about making http requests, I added this block to prevent anyone from adding any tests in here in the future that make actual http requests, since that would make the test slow and flaky. Just future proofing.
But considering that I already added this in the `setup` script in `core` tests, this is indeed unnecessary. I'll remove it.